lib/block_view.go

```diff
@@ -1068,22 +1070,32 @@ func (bav *UtxoView) CopyUtxoView() (*UtxoView, error) {
 }
 
 func NewUtxoView(
-	_handle *badger.DB, _params *BitCloutParams, _bitcoinManager *BitcoinManager) (*UtxoView, error) {
+	_handle *badger.DB,
+	_params *BitCloutParams,
+	_bitcoinManager *BitcoinManager,
+	_postgres *Postgres,
+) (*UtxoView, error) {
 
 	view := UtxoView{
 		Handle:         _handle,
+		Postgres:       _postgres,
 		Params:         _params,
 		BitcoinManager: _bitcoinManager,
-		// Note that the TipHash does not get reset as part of
-		// _ResetViewMappingsAfterFlush because it is not something that is affected by a
-		// flush operation. Moreover, its value is consistent with the view regardless of
-		// whether or not the view is flushed or not. Additionally the utxo view does
-		// not concern itself with the header chain (see comment on GetBestHash for more
-		// info on that).
-		TipHash: DbGetBestHash(_handle, ChainTypeBitCloutBlock /* don't get the header chain */),
-
 		// Set everything else in _ResetViewMappings()
 	}
+
+	// Note that the TipHash does not get reset as part of
+	// _ResetViewMappingsAfterFlush because it is not something that is affected by a
+	// flush operation. Moreover, its value is consistent with the view regardless of
+	// whether or not the view is flushed or not. Additionally the utxo view does
+	// not concern itself with the header chain (see comment on GetBestHash for more
+	// info on that).
+	if view.Postgres != nil {
+		view.TipHash = view.Postgres.GetChain("main").TipHash
+	} else {
+		view.TipHash = DbGetBestHash(view.Handle, ChainTypeBitCloutBlock /* don't get the header chain */)
+	}
```

Comment:
Shouldn't the node depend on the badger tip hash more than the postgres tiphash?  I don't think we want to depend on things in the postgres instance, even if it is available.